frontend/src/metabase/dashboard/actions/parameters.ts

```diff
@@ -115,30 +148,35 @@ export const SET_PARAMETER_MAPPING = "metabase/dashboard/SET_PARAMETER_MAPPING";
 
 export const setParameterMapping = createThunkAction(
   SET_PARAMETER_MAPPING,
-  (parameter_id, dashcard_id, card_id, target) => {
+  (
+    parameterId: ParameterId,
+    dashcardId: DashCardId,
+    cardId: CardId,
+    target: ParameterTarget | null,
+  ) => {
     return (dispatch, getState) => {
       dispatch(closeAutoWireParameterToast());
 
-      const dashcard = getDashCardById(getState(), dashcard_id);
+      const dashcard = getDashCardById(getState(), dashcardId);
 
       if (target !== null) {
         dispatch(
           autoWireDashcardsWithMatchingParameters(
-            parameter_id,
-            dashcard,
+            parameterId,
+            dashcard as QuestionDashboardCard,
```

Comment:
@kulyk please have a look, maybe you have a better idea of how to deal with the type mismatch?

Reply:
You might want to narrow down `dashcard's` type before doing the mapping. You can find a type-guard per each subtype of a daschard in `metabase/dashboard/utils` (`isQuestionDashCard`, `isActionDashCard` and `isVirtualDashCard`)